fetch.bs

```diff
@@ -1170,45 +1170,56 @@ or "<code>worker</code>".
+promise <var>p</var>, run these steps <a>in parallel</a>:
 
 <ol>
- <li>Let <var>body</var> be <var>request</var>'s <a for=request>body</a>.
-
- <li><p>If <var>body</var> is null, then <a>queue a fetch task</a> on <var>request</var> to
- <a>process request end-of-body</a> for <var>request</var> and abort these steps.
+ <li><p>Wait for <var>p</var> to be fulfilled or rejected.
 
  <li>
-  <p>Let <var>read</var> be the result of <a lt="read a chunk" for=ReadableStream>reading a
-  chunk</a> from <var>body</var>'s <a for=body>stream</a>.
+  <p>If <var>p</var> is fulfilled with an object whose <code>done</code> property is false
+  and whose <code>value</code> property is a <code>Uint8Array</code> object, run these steps:
 
-  <ul>
-   <li>
-    <p>When <var>read</var> is fulfilled with an object whose <code>done</code>
-    property is false and whose <code>value</code> property is a
-    <code>Uint8Array</code> object, run these substeps:
+   <ol>
+    <li><p>Let <var>bs</var> be the <a>byte sequence</a> represented by the <code>Uint8Array</code>
+    object.
 
-    <ol>
-     <li><p>Let <var>bytes</var> be the <a>byte sequence</a> represented by the
-     <code>Uint8Array</code> object.
+    <li>
+     <p>Transmit <var>bs</var>. Whenever one or more bytes are transmitted, increase
+     <var>body</var>'s <a for=body>transmitted bytes</a> by the number of transmitted bytes and
+     <a>queue a fetch task</a> on <var>request</var> to <a>process request body</a>
+     for <var>request</var>.
 
-     <li><p>Transmit <var>bytes</var>.
+     <p class="note no-backref">This step blocks until <var>bs</var> is fully transmitted.
 
-     <li><p>Increase <var>body</var>'s <a for=body>transmitted bytes</a> by <var>bytes</var>'s
-     length.
+    <li><p>Let <var>read</var> be the result of <a lt="read a chunk" for=ReadableStream>reading a
+    chunk</a> from <var>body</var>'s <a for=body>stream</a>.
 
-     <li><p>Run the above step again.
-    </ol>
+    <li><p><a for=request>Transmit a chunk</a> with <var>read</var>.
+   </ol>
 
-   <li><p>When <var>read</var> is fulfilled with an object whose <code>done</code>
-   property is true, <a>queue a fetch task</a> on <var>request</var> to
-   <a>process request end-of-body</a> for <var>request</var>.
+ <li><p>If <var>p</var> is fulfilled with an object whose <code>done</code> property is true,
+ <a>queue a fetch task</a> on <var>request</var> to <a>process request end-of-body</a> for
+ <var>request</var>.
 
-   <li><p>When <var>read</var> is fulfilled with a value that matches with neither of the
-   above patterns, or <var>read</var> is rejected, <a lt=terminated for=fetch>terminate</a> the
-   ongoing fetch with reason <i>fatal</i>.
-  </ul>
+ <li><p>If <var>p</var> is fulfilled with a value that matches with neither of the above patterns,
+ or <var>read</var> is rejected, <a lt=terminated for=fetch>terminate</a> the ongoing fetch with
```

Comment:
"or read is rejected" isn't needed here, right? That already follows from p being fulfilled with a "wrong" value.